### PR TITLE
build: Support lerna --scope filter flag (#4484)

### PR DESCRIPTION
**Pull request by jd-carroll**
Support the lerna `--scope` filter flag

### CODE REVIEW

**Comment by jd-carroll** [comment]
Feature already exists in `scripts/build[up|down]`

**Comment by jd-carroll** [comment]
Reopening as I think incorporating the `buildup` / `builddown` into the regular build script provides value (fixes #4484)

File: build.sh [hunk 1]
```diff
@@ -18,6 +22,19 @@ while [[ "${1:-}" != "" ]]; do
         --skip-test|--skip-tests)
             runtarget="build"
             ;;
+        --scope)
```
Comment (eladb) [inline build.sh]: Can we default scope to be the package in the current directory in case this script is executed from a non-root directory?

Comment (jd-carroll) [reply build.sh]: @eladb Great idea!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -199,12 +199,18 @@ The build process is divided into stages, so you can invoke them as needed:
...
 
-### Partial build tools
+##### Partial builds
 
-There are also two useful scripts in the `scripts` directory that can help you build part of the repo:
+When working on an individual module, the `build.sh` can be used to build that module specifically along with any related modules. The following is a list of scripts to build a module under development:
 
-- __`scripts/buildup`__: builds the current module and all of its dependencies (in topological order).
-- __`scripts/builddown`__: builds the current module and all of its consumers (in topological order).
+- __`build.sh --scope <module-name>`__: builds the module with the specified name.
```
Comment (eladb) [inline CONTRIBUTING.md]: Add `[]` around scope to indicate it's optional and mention that it defaults to the current module or the entire project when executed from the repo root

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -366,8 +372,7 @@ In many cases, you don't really need to build the entire project. Say you want t
 
 ```console
 $ ./install.sh
-$ cd packages/@aws-cdk/aws-ec2
-$ ../../../scripts/buildup
+$ ./build --scope @aws-cdk/aws-ec2 --skip-test --skip-api-check
```
Comment (eladb) [inline CONTRIBUTING.md]: `shell
$ cd packages/@aws-cdk/aws-ec2
$ ../../../build.sh --skip-test --skip-api-check
```

File: build.sh [hunk 4]
```diff
@@ -2,11 +2,16 @@
...
+up=""
+down=""
+skipapi=false
+skipclean=false
 runtarget="build+test"
 while [[ "${1:-}" != "" ]]; do
     case $1 in
         -h|--help)
-            echo "Usage: build.sh [--no-bail] [--force|-f] [--skip-test]"
+            echo "Usage: build.sh [--no-bail] [--force|-f] [--skip-test] [--scope <package-name> [--up] [--down]] [--skip-api-check] [--skip-clean]"
```
Comment (eladb) [inline build.sh]: can we alias `--up` and `--down` with `-u` and `-d`

File: CONTRIBUTING.md (hunk 2 above)
Comment (jd-carroll) [reply CONTRIBUTING.md]: Indicated all of the `<module-name>` references are optional

**Comment by jd-carroll** [comment]
I updated the docs, see if that will work. Part of my objective for this PR was to avoid `cd` 'ing out of the top level project dir.  However, as I've gone through all of the related build scripts (which was partly the point), I've come to realize just how much tooling has been built around `cd` 'ing into the module directory. In some ways I've begun to question whether this PR is pushing the project in the right direction...

[TLDR] If you feel enabling dev's to stay in the top level project dir, then this PR is a good step in that direction.  If you do not feel that working from the module directory is the right direction, then I would say we should close and not merge this PR.

**Comment by jd-carroll** [comment]
Note: The build failure is unrelated to the changes in this PR

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -199,12 +199,18 @@ The build process is divided into stages, so you can invoke them as needed:
...
 
-### Partial build tools
+##### Partial builds
 
-There are also two useful scripts in the `scripts` directory that can help you build part of the repo:
+When working on an individual module, the `build.sh` can be used to build that module specifically along with any related modules. The following is a list of scripts to build a module under development:
 
-- __`scripts/buildup`__: builds the current module and all of its dependencies (in topological order).
-- __`scripts/builddown`__: builds the current module and all of its consumers (in topological order).
+- __`build.sh --scope [<module-name>]`__: builds the module with the specified name.
```
Comment (eladb) [inline CONTRIBUTING.md]: Please indicate that `--scope` is optional if your working directory is a specific module. I think that would be the more common use case for most people:

```shell
build.sh --up [-- scope MODULE]
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (eladb) [reply CONTRIBUTING.md]: Okay, I am fine with this too

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -364,10 +370,17 @@ $ docker run -v $(pwd):/app -w /app aws-cdk <CDK ARGS>
...
+$ ./build --scope @aws-cdk/aws-ec2 --skip-test --skip-api-check
+```
+
+Alternatively, you can also build from the directory containing the `@aws-cdk/aws-ec2` module:
+
 ```console
 $ ./install.sh
 $ cd packages/@aws-cdk/aws-ec2
-$ ../../../scripts/buildup
+$ ./build --scope --skip-test --skip-api-check
```
Comment (eladb) [inline CONTRIBUTING.md]: `--scope`? I expected `--scope` to not be required in this case

**Comment by eladb** [comment]
@jd-carroll any updates here? Should we close this for now?

**Pull request closed without merging**